R/fcast.R

```diff
@@ -8,9 +8,21 @@ guess = function(x) {
   return(var)
 }
 
-dcast = function(data, formula, fun.aggregate = NULL, ..., margins = NULL,
-                  subset = NULL, fill = NULL, value.var = guess(data)) {
-  UseMethod("dcast", data)
+dcast <- function(
+  data, formula, fun.aggregate = NULL, ..., margins = NULL,
+  subset = NULL, fill = NULL, value.var = guess(data)
+) {
+  if (is.data.table(data)) UseMethod("dcast", data)
+  # nocov start
+  else {
+    # reshape2::dcast is not generic so we have to call it explicitly. See comments at the top of fmelt.R too.
+    ns = tryCatch(getNamespace("reshape2"), error=function(e)
```

Comment:
why `getNamespace` rather than `requireNamespace`? then no `tryCatch` is needed

Reply:
is it related to removing `reshape2` from suggests? if so then it looks hacky

Reply:
this is simply reversion to the old code :)

Reply:
https://github.com/Rdatatable/data.table/blob/1ae2a5d75cd53349c98b3eccedf8f91413c08b81/R/fcast.R#L15-L23

Reply:
https://github.com/Rdatatable/data.table/commit/0acf022d087ed996516a6eb788e6da383dcb1a81
yes